registry_test.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import "testing"
+
+func Test_GetAuthURL(t *testing.T) {
+	tcs := []struct {
+		challenge     string
+		expectedURL   string
+		expectedError bool
+		description   string
+	}{
+		{
+			description: "base case",
```

Comment:
Nitpick but could we use the "case 0:" style we have in `fmt`? 

I find it quicker to track down test failures.

Reply:
Sure, done.